src/DiIiS-NA/D3-GameServer/GSSystem/ActorSystem/Implementations/Hirelings/Hireling.cs

```diff
@@ -530,18 +530,23 @@ public void RefreshEquipment(Player player)
 
 		#region EqupimentStats
 
-		public List<Item> GetEquippedItems(Player player)
+		public IEnumerable<Item> GetEquippedItems(Player player)
 		{
-			return _equipment[player].Values.ToList();
+			return _equipment[player].Values;
 		}
 
 		public float GetItemBonus(GameAttributeF attributeF)
 		{
-			var stats = GetEquippedItems(owner).Where(item => item.Attributes[GameAttribute.Durability_Cur] > 0 || item.Attributes[GameAttribute.Durability_Max] == 0);
+			var stats = GetEquippedItems(owner)
+				.Where(item => item.Attributes[GameAttribute.Durability_Cur] > 0 ||
+				               item.Attributes[GameAttribute.Durability_Max] == 0);
 
 			if (attributeF == GameAttribute.Attacks_Per_Second_Item)
-				return stats.Count() > 0 ? stats.Select(item => item.Attributes[attributeF]).Where(a => a > 0f).Aggregate(1f, (x, y) => x * y) : 0f;
-
+			{
+				return stats.Any()
+					? stats.Select(item => item.Attributes[attributeF]).Where(a => a > 0f).Aggregate(1f, (x, y) => x * y)
```

Comment:
Here I have a question.
What if we have something in stats, but all values < 0? In this case `stats.Any()` will return true, but `stats.Select(item => item.Attributes[attributeF]).Where(a => a > 0f)` will return empty enumeration and finally `Aggregate` will return 1. Is this expected behavior?